armi/reactor/reactors.py

```diff
@@ -2129,9 +2129,83 @@ def growToFullCore(self, cs):
+            gridPlateMaterial = gridPlateComponent.material
+            if not isinstance(gridPlateMaterial, type(refGridPlateMat)):
+                runLog.warning(
+                    "Grid plate material for assembly is not consistent! Grid plate expansion/contraction assumes\n"
+                    f"a consistent material and will use {refGridPlateMat}. The inconsistency is:\n"
+                    f"Reference grid plate material = {refGridPlateMat}\n"
+                    f"Inconsistent grid plate material = {gridPlateMaterial} -- is in {b.parent}"
+                )
+
+    def calculateNewGridPlatePitchFromTemp(self, inletTemp: float) -> float:
+        """calculate a grid plate pitch using a new bulk ave core inlet temperature in C
+
+        Parameters
+        ----------
+        inletTemp : float
+            an isothermal temperature in C to apply to cs["Tin"]
         """
-        set the pitch in all blocks
+        self.checkGridPlateMaterialConsistency()
+        # retrieve the first grid plate's material (all grid plates are the same material, so just get the first)
+        gridPlateMaterial = (
+            self.getFirstBlock(Flags.GRID_PLATE).getComponent(Flags.GRID_PLATE).material
+        )
+        currentGridPlatePitch = self.getAssemblyPitch()
+        currentGridPlateTemp = self.refAssem.p.THcoolantInletT
+        runLog.info(
+            f"Using a {Flags.GRID_PLATE} material of {gridPlateMaterial} for grid plate expansion/contraction calculations "
+            f"with an initial temperature of {currentGridPlateTemp} C and new temperature of {inletTemp} C."
+        )
+        # compute expansion/contraction fraction
+        dll = gridPlateMaterial.linearExpansionFactor(
+            Tc=inletTemp,
+            T0=currentGridPlateTemp,
+        )
+        # compute new grid place pitch
+        return currentGridPlatePitch * (1.0 + dll)
+
+    def updateGridPlatePitch(self, inletTemp: float, updateInletTemp: bool = False):
+        """update the grid plate pitch using a new bulk ave core inlet temperature in C
+
+        Parameters
+        ----------
+        inletTemp : float
+            an isothermal core bulk average inlet temperature in C
+        updateInletTemp: bool, optional
+            optional boolean to determine if the bulk average core inlet temperature should be updated to inletTemp
+
+        Notes
+        -----
+        1) all grid plate blocks are the same material
+        2) all grid plate blocks are isothermal and equal to cs["Tin"]
+        """
+        newGridPlatePitch = self.calculateNewGridPlatePitchFromTemp(inletTemp)
+        # update the pitch
+        self.setPitchUniform(newGridPlatePitch)
+        if updateInletTemp:
+            self.updateInletTemp(inletTemp)
+
+    def updateInletTemp(self, inletTemp):
+        """update core bulk average inlet temp for all assemblies"""
+        for a in self.getAssemblies():
+            a.p.THcoolantInletT = inletTemp
```

Comment:
See above comment on having an assembly method to reuse.

Reply:
Addressed https://github.com/terrapower/armi/pull/993/commits/abfaede115874ce52ead2cf2f222640f0d5d4d12